packages/react-sdk/src/components/button.ws.tsx

```diff
@@ -14,7 +14,7 @@ const presetStyle = {
 
 export const meta: WsComponentMeta = {
   category: "general",
-  type: "container",
+  type: "embed",
```

Comment:
this will prevent user from having an icon inside of a button

Reply:
I think we need a more hollistic approach to this problem, to look more generally at how we would prevent nesting wrong elements inside other wrong elements

Reply:
Since, we the studio always knows the elements that are supported in the platform. How about every element have a new attribute that says, what `type` of children are allowed to add 🤔 
Some element to element mappers of sort ?

Reply:
Good catch btw 👍

Reply:
We have been discussing this internally, @TrySound has something to say about it

Reply:
Describing all possible types might not be a good idea because the list can grow a lot. Maybe support forbiddenDescendants?

Reply:
Or even forbiddenAncestors? Easier to write.

Reply:
yes, but should we have a global map of forbidden descendants and forbidden ancestors?

Reply:
Yes, i am also thinking along those lines. Should i go ahead and try something out ?

Reply:
No worries I already started

Reply:
Ah, ok ok. I had some thing like this and use them in return to type the map and props meta and elements 😄 

```ts
export enum ELEMENTS {
  SLOT = 'Slot',
  FRAGMENT = 'Fragment',
  BOX = 'Box',
  BODY = 'Body',
  TEXT_BLOCK = 'Text Block',
  HEADINNG = 'Heading',
  PARAGRAPH = 'Paragraph',
  LINK = "Link",
  LINK_BLOCK = 'Link Block',
  RICH_TEXT_LINK = 'Link Text',
  SPAN = 'Span',
  BOLD = 'Blod',
  ITALIC = 'Italic',
  SUPERSCRIPT = 'Superscript',
  SUBSCRIPT = 'Subscript',
  BUTTON = 'Button',
  INPUT = 'Input',
  FORM = 'Form',
  IMAGE = 'Image',
  BLOCKQUOTE = 'Blockquote',
  LIST = 'List',
  LIST_ITEM = 'List Item',
  SEPERATOR = 'Separator',
  CODE = 'Code',
  LABEL = 'Label',
  SUCCESS_MESSAGE = 'Success Message',
  ERROR_MESSAGE = 'Error Message',
  TEXT_AREA = 'Text Area',
  RADIO_BUTTON_FIELD = "Radio Button Field",
  RADIO_BUTTON = 'Radio Button',
  CHECKBOX_FIELD = 'Checkbox Field',
  CHECKBOX = 'Checkbox'
}

```